### ISSUE TITLE
Enhancement: Q-Learning reinforcement router for agent task routing

### ISSUE THREAD

**Issue by mrveiss** [mrveiss/AutoBot-AI#2092]
## Summary

Replace the LLM-based complex routing fallback in `AgentRouter` with a Q-Learning reinforcement learning router that learns optimal agent selection from task outcomes — eliminating one LLM API call per routed request while improving routing accuracy over time.

## Motivation

- Current `AgentRouter.determine_routing()` calls an LLM (temperature 0.3) when quick pattern matching confidence is ≤0.8 — this is expensive and adds latency
- Inspired by Q-Learning routing patterns from multi-agent orchestration research (epsilon-greedy, experience replay, feature-based state hashing)
- AutoBot already has `task_pattern_learner.py` (#930) that records task outcomes — this data can bootstrap the Q-table
- The existing `TieredModelRouter` (#748) uses heuristic scoring — an RL approach can learn optimal tier selection from actual outcomes

## Proposed Architecture

```
Request → Quick Pattern Match (existing, fast)
  ├─ confidence > 0.8 → use quick route (unchanged)
  └─ confidence ≤ 0.8 → Q-Learning Router (new, <1ms)
       ├─ confidence > 0.6 → use RL route
       └─ confidence ≤ 0.6 → LLM fallback (existing, last resort)
```

**Key components (Python, Redis-backed):**
- Feature-based state hashing: keyword detection + query length + n-gram hash → state key
- Epsilon-greedy action selection with cosine decay (explore → exploit)
- Experience replay buffer in Redis list (prioritized by TD-error)
- Q-table persisted in Redis hash (`rl:router:qtable:{state}`)
- Reward signal from existing task outcome tracking

## Files Affected

- `autobot-backend/agents/agent_orchestration/rl_router.py` (new) — Q-Learning router implementation
- `autobot-backend/agents/agent_orchestration/routing.py` — integrate RL router between pattern match and LLM fallback
- `autobot-backend/agents/task_pattern_learner.py` — wire outcome data as reward signal to RL router

## Dependencies & Sequencing

- **No blockers** — this is additive, sits between existing quick-match and LLM fallback
- **Does NOT touch** Neural Mesh RAG (#1994), search components, or reranking
- **Complements** TieredModelRouter — could later share the RL backbone for model tier selection
- **Bootstrap data**: use existing `task_pattern_learner` outcome history to pre-train Q-table

## Risk Mitigation

- RL router starts with high epsilon (exploration mode) — effectively random initially, falling back to LLM
- As it learns, epsilon decays and RL handles more requests autonomously
- LLM fallback always available as safety net
- Kill switch: set `rl_routing_enabled=False` in config to revert to pure LLM routing
- Q-table reset via admin endpoint if routing degrades

## Acceptance Criteria

- [ ] Q-Learning router with epsilon-greedy selection, experience replay, Redis persistence
- [ ] Integrated into AgentRouter between quick-match and LLM fallback
- [ ] Reward signal wired from task outcome tracking
- [ ] Config: `rl_routing_enabled`, `epsilon_initial`, `epsilon_final`, `learning_rate`, `gamma`
- [ ] Metrics: routing decisions/sec, RL vs LLM fallback ratio, avg confidence, Q-table size
- [ ] Unit tests for: routing, learning, persistence, fallback behavior
- [ ] No regression when disabled
